tests/project/test_config.py

```diff
@@ -46,36 +46,41 @@ def mock_getenv(key: str, default: Optional[str] = None) -> Optional[str]:
             # a definition structure for these values but directly return defaults
             # in "getter" functions (higher-level data structures).
             local = generate_local_override_yml(project)
-            assert local["native_app"]["application"]["name"] == "minimal_jsmith"
-            assert local["native_app"]["application"]["role"] == "resolved_role"
-            assert (
-                local["native_app"]["application"]["warehouse"] == "resolved_warehouse"
-            )
-            assert local["native_app"]["application"]["debug"] == True
-            assert local["native_app"]["package"]["name"] == "minimal_pkg_jsmith"
-            assert local["native_app"]["package"]["role"] == "resolved_role"
+            assert local.native_app.application.name == "minimal_jsmith"
+            assert local.native_app.application.role == "resolved_role"
+            assert local.native_app.application.warehouse == "resolved_warehouse"
+            assert local.native_app.application.debug == True
+            assert local.native_app.package.name == "minimal_pkg_jsmith"
+            assert local.native_app.package.role == "resolved_role"
 
 
 @pytest.mark.parametrize("project_definition_files", ["underspecified"], indirect=True)
 def test_underspecified_project(project_definition_files):
-    with pytest.raises(YAMLValidationError) as exc_info:
+    with pytest.raises(ValidationError) as exc_info:
         load_project_definition(project_definition_files)
 
-    assert "required key(s) 'artifacts' not found" in str(exc_info.value)
+    assert (
+        "Field required [type=missing, input_value={'name': 'underspecified'}, input_type=dict]"
+        in str(exc_info.value)
+    )
 
 
 @pytest.mark.parametrize(
     "project_definition_files", ["no_definition_version"], indirect=True
 )
 def test_fails_without_definition_version(project_definition_files):
-    with pytest.raises(YAMLValidationError) as exc_info:
+    with pytest.raises(ValidationError) as exc_info:
         load_project_definition(project_definition_files)
 
-    assert "required key(s) 'definition_version' not found" in str(exc_info.value)
+    assert "definition_version" in str(exc_info.value)
```

Comment:
Let's make longer assertion so we can see the error

Reply:
Done